packages/docusaurus-plugin-pwa/package.json

```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@docusaurus/plugin-pwa",
+  "version": "2.0.0-alpha.50",
+  "description": "Docusaurus Plugin to add PWA support",
+  "main": "src/index.js",
+  "publishConfig": {
+    "access": "public"
+  },
+  "license": "MIT",
+  "dependencies": {
+    "webpack-merge": "^4.2.2",
+    "webpack": "^4.41.2",
+    "workbox-build": "^5.1.2",
```

Comment:
Any reason this isn't just `workbox-webpack-plugin` (with activation in `Plugin.configureWebpack`)?

Reply:
@RDIL a few reasons actually:

1. This is similar to what vue-press does for their PWA plugin:https://git.io/JfLqz
2. I wanted the ability to process the SW source file through Webpack. That way, developers can leverage whatever NPM dependencies they need in their SW
3. Also, using the Webpack plugins, I had to import the entire Workbox runtime. By using Webpack to process the SW, I was able to import only the Workbox dependencies I need, like `workbox-precaching`
4. If I wanted to build the SW with Webpack, and then use `workbox-webpack-plugin` in `configureWebpack()`, I would have to build the SW first, then build the client and server. Right now, there's no [`preBuild()` lifecycle](https://v2.docusaurus.io/docs/lifecycle-apis), so I did it in `postBuild()`
5. Since Docusaurus is [server rendered](https://docusaurus.io/blog/2018/09/11/Towards-Docusaurus-2#content), there are two Webpack configs and therefore two manifests. This means the service worker has to import both manifests. The reason why is because JS and CSS assets are generated by the client Webpack compiler, while HTML assets are generated by the server (I think). 
    - Also if I remember correctly, there was an instance where some server files were included in the manifest, but not part of the actual output, so I had to do some special filtering. By using the CLI and glob pattern directly, it generates a **single** manifest that corresponds to all the files physically present in the filesystem, which gives me more confidence than filtering out files that don't work for some reason